site/zenodo_rdm/legacy/deserializers/schemas.py

```diff
@@ -238,11 +248,203 @@ def _additional_descriptions(self, data):
         """Transform notes of a legacy record."""
         if not data:
             return None
+        rdm_additional_descriptions = []
+
+        notes = data.get("notes")
+        if notes:
+            rdm_additional_descriptions.append(
+                {"description": notes, "type": {"id": "other"}}
+            )
 
-        rdm_additional_descriptions = [{"description": data, "type": {"id": "other"}}]
+        method = data.get("method")
+        if method:
+            rdm_additional_descriptions.append(
+                {"description": method, "type": {"id": "methods"}}
+            )
 
         return rdm_additional_descriptions
 
+    def _locations(self, data):
+        """Transform locations of a legacy record."""
+        if not data:
+            return None
+
+        features = []
+        for legacy_location in data:
+            lat = legacy_location.get("lat")
+            lon = legacy_location.get("lon")
+            place = legacy_location["place"]
+            description = legacy_location.get("description")
+
+            feature = {"place": place}
+
+            if lat and lon:
+                geometry = {"type": "Point", "coordinates": [lon, lat]}
+                feature.update({"geometry": geometry})
+
+            if description:
+                feature.update({"description": description})
+
+            features.append(feature)
+
+        return {"features": features}
+
+    def _subjects(self, data):
+        """Transform subjects of a legacy record.
+
+        RDM subjects translate to either legacy keywords or subjects.
+        """
+
+        def _subjects_from_keywords(keywords):
+            """Legacy keywords are free text strings.
+
+            They map to custom subjects.
+            """
+            return [{"subject": kw} for kw in keywords]
+
+        def _subjects_from_subjects(data):
```

Comment:
naming: a bit redundant, since they are inside `_subjects` what about `_from_keywords` and `_from_subjects`